### ISSUE TITLE
Removal of the data refresh server

### ISSUE THREAD

**Issue by stacimc** [WordPress/openverse#3925]
| Start Date        | ETA | Project Lead | Actual Ship Date |
| ---------- | ---------- | ------------ | ---------------- |
| 2024-03-14 |  | @stacimc  | TBD              |

## Description

We presently orchestrate our data refresh process with Airflow, but the operations themselves occur on a bespoke data refresh server that’s difficult to maintain, troublesome to deploy, and not nearly as robust to failures.

For this project, we will  move the operational pieces of the data refresh into Airflow.

## Documents

Project proposal
Implementation Plan

## Issues

- 

## Milestones

-

**Comment by stacimc** [comment]
Adding myself as project lead at least for now. As discussed at the priority meeting, we will move forward with making the proposal and at least starting work on the implementation plan, with the understanding that implementation may not move forward if this project is discovered to be more complicated than we believe it will be.

**Comment by stacimc** [comment]
Updated the language from "data refresh server" to "ingestion server" to reflect the current name of the service, in order to be less confusing. We had experimented with renaming the ingestion server in some places in documentation; if this project proceeds we won't have to worry about this confusion anymore regardless :)

Work on the project proposal is underway, as is some early investigation into the feasibility of the project.

**Comment by stacimc** [comment]
The project proposal was approved, and the [IP](https://github.com/WordPress/openverse/pull/4026) is awaiting its second approval.

**Comment by stacimc** [comment]
The implementation plan has been merged and approved, and issues created under the milestone.

**Comment by zackkrida** [comment]
@stacimc I've moved this to "On Hold" while we wait to determine how to prioritize it; whether we start work on this project or aim to complete others first.

**Comment by stacimc** [comment]
After discussing with @zackkrida we've decided to move this into `In Progress` and kick off implementation. I'll be picking up the first issue today 🥳

**Comment by stacimc** [comment]
This project is underway. The [first major PR](https://github.com/WordPress/openverse/pull/4259) is up for review, with work in progress on the indexer worker image. There is also considerable progress on the [infrastructure](https://github.com/WordPress/openverse-infrastructure/pull/886) side with preparing for the indexer workers.

**Comment by stacimc** [comment]
Progress was delayed due to AFK. The indexer worker has been drafted [here](https://github.com/WordPress/openverse/pull/4330) and will be up for review this week.

**Comment by stacimc** [comment]
The indexer worker is fully implemented and we're moving on to the last few implementation issues. One [additional issue](https://github.com/WordPress/openverse/issues/4542) was added to the milestone after discussion on #4464.

**Comment by stacimc** [comment]
After a conversation with @sarayourfriend yesterday, I'm considering removing the use of the autoscaling group and going back to a plan very similar to my original proposal in the IP for this project, allowing Airflow to directly manage the EC2 instances. The ASG has caused a few problems with error handling and, as noted in [inline comments in the PR for implementing the distributed reindex](https://github.com/WordPress/openverse/pull/4572/files#diff-e12e3a9203428649ff02d93be42905cd7c34e920e7233f929b6b9411aa801dacR171), with retrying individual workers (one of the stated goals of the project, and a situation we've run into in recent memory).

This is not yet definitive and I will make a PR to change the implementation plan since this is a big enough change/I want to get approval. However I'll note that this would not invalidate any work that's been done so far, except for a few in-progress changes in the linked PR (which need work either way!), and removing just the ASG on the infrastructure side.

**Comment by sarayourfriend** [comment]
Great! I'll look out for the IP change (please ping me there) and will get the infra side ready for you, it should be a simple enough change, with some room for a small refactor I've been wanting to do for a while now (extract the launch template and security group creation out into a separate module, so that one-off instances like the bastion can use it instead of the old user-data approach. Anyway, it should be a quick one to implement on the infra side and unblock live testing with staging.

**Comment by stacimc** [comment]
This project was considerably delayed due to a sequence of AFK and assigned support work/meetups. Work has now been resumed. Importantly the following have been merged:
* [Adding the distributed reindex steps](https://github.com/WordPress/openverse/pull/4572/)
* [Adding the filter data steps](https://github.com/WordPress/openverse/issues/4541)

The final large chunk of implementation, to [add the remaining steps](https://github.com/WordPress/openverse/issues/4149), is in progress. Afterward there will be a few cleanup PRs and small pieces, but the major piece of work left will be integration tests.

**Comment by stacimc** [comment]
This project is code complete and has just been postponed awaiting time to rigorously test the new data refresh in staging. Any issues discovered during testing will then have to be addressed, and the project will be kept open for an extended period of time until several production runs have been completed before we retire the old dags.

**Comment by stacimc** [comment]
After tackling a number of small issues encountered during testing, the staging audio data refresh has been run successfully on the production Airflow instance, and the staging image data refresh is underway!
